From d308c6b539628317add8c4d54d06ab2e1efe5de4 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 28 May 2025 10:41:56 +0100
Subject: [PATCH] Updates for OE cross builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 src/emitterutils.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/emitterutils.cpp b/src/emitterutils.cpp
index c6ad5e5..ab4c750 100644
--- a/src/emitterutils.cpp
+++ b/src/emitterutils.cpp
@@ -1,6 +1,7 @@
 #include <algorithm>
 #include <iomanip>
 #include <sstream>
+#include <cstdint>
 
 #include "emitterutils.h"
 #include "exp.h"
-- 
2.34.1

