From 44e13ab78a43419c9e60d24d4955f8a24798eb7d Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Tue, 24 Feb 2026 09:24:17 -0800
Subject: [PATCH] Fix FMT installation

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 cmake/Dependencies.cmake | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 903c212de..04e501505 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -1587,7 +1587,9 @@ endif()
 # This was the default behavior before version 12.0.0.
 # Since PyTorch C API depends on it, make it available for projects that
 # depend on PyTorch.
-set(FMT_INSTALL ON)
+if(NOT DEFINED FMT_INSTALL)
+  set(FMT_INSTALL ON)
+endif()
 set(TEMP_BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS})
 set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared libs" FORCE)
 add_subdirectory(${PROJECT_SOURCE_DIR}/third_party/fmt)
-- 
2.43.0

