From 536fa196282cb5aee2622765e47ccc959717b9e4 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 6 May 2026 12:46:18 -0700
Subject: [PATCH] update CPM version

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 cmake/CPM.cmake | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/cmake/CPM.cmake b/cmake/CPM.cmake
index 2d151bf..3636ee5 100644
--- a/cmake/CPM.cmake
+++ b/cmake/CPM.cmake
@@ -45,7 +45,7 @@ endif()
 if(DEFINED EXTRACTED_CPM_VERSION)
   set(CURRENT_CPM_VERSION "${EXTRACTED_CPM_VERSION}${CPM_DEVELOPMENT}")
 else()
-  set(CURRENT_CPM_VERSION 1.0.0-development-version)
+  set(CURRENT_CPM_VERSION 0.42.0)
 endif()
 
 get_filename_component(CPM_CURRENT_DIRECTORY "${CMAKE_CURRENT_LIST_DIR}" REALPATH)
-- 
2.43.0

