From fcc6c2b7974a42ac8f9466e1cd8e29bb4734c9c6 Mon Sep 17 00:00:00 2001
From: Matt Madison <matt@madison.systems>
Date: Sat, 19 Oct 2024 08:02:15 -0700
Subject: [PATCH] OE cross-build fixups

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Matt Madison <matt@madison.systems>
---
 cuda/setup.py | 15 ++-------------
 1 file changed, 2 insertions(+), 13 deletions(-)

diff --git a/cuda/setup.py b/cuda/setup.py
index ec52362..12565e5 100644
--- a/cuda/setup.py
+++ b/cuda/setup.py
@@ -155,26 +155,15 @@ for file in input_files:
 # Prepare compile arguments
 
 # For Cython
-include_dirs = [
-    os.path.dirname(sysconfig.get_path("include")),
-] + include_path_list
-library_dirs = [sysconfig.get_path("platlib"), os.path.join(os.sys.prefix, "lib")]
+include_dirs = []
+library_dirs = []
 
 extra_compile_args = []
 extra_cythonize_kwargs = {}
 if sys.platform != 'win32':
     extra_compile_args += [
-        '-std=c++14',
         '-fpermissive',
-        '-Wno-deprecated-declarations',
-        '-D _GLIBCXX_ASSERTIONS',
-        '-fno-var-tracking-assignments'
     ]
-    if '--debug' in sys.argv:
-        extra_cythonize_kwargs['gdb_debug'] = True
-        extra_compile_args += ['-g', '-O0']
-    else:
-        extra_compile_args += ['-O3']
 
 # For Setup
 extensions = []
