From 487abc42f0080f3dac1e4908857d913298cc7a4b Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ilies.chergui@gmail.com>
Date: Thu, 10 Nov 2022 19:33:38 +0000
Subject: [PATCH] Build fixups

Upstream-Status: Inappropriate [OE-Specific]
Signed-off-by: Ilies CHERGUI <ilies.chergui@gmail.com>
Signed-off-by: Roger Knecht <roger@norberthealth.com>
Signed-off-by: Matt Madison <matt@madison.systems>
---
 CMakeLists.txt | 27 ++++++---------------------
 1 file changed, 6 insertions(+), 21 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 622f69a..de10095 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -45,30 +45,17 @@ endif()
 #
 # Dependencies
 #
-include(FetchContent)
-
-FetchContent_Declare(
-  repo-common
-  GIT_REPOSITORY https://github.com/triton-inference-server/common.git
-  GIT_TAG ${TRITON_COMMON_REPO_TAG}
-  GIT_SHALLOW ON
-)
-FetchContent_Declare(
-  repo-core
-  GIT_REPOSITORY https://github.com/triton-inference-server/core.git
-  GIT_TAG ${TRITON_CORE_REPO_TAG}
-  GIT_SHALLOW ON
-)
-FetchContent_MakeAvailable(repo-common repo-core)
 
 #
 # CUDA
 #
 if(${TRITON_ENABLE_GPU})
-  find_package(CUDAToolkit REQUIRED)
+  enable_language(CUDA)
   find_package(CUDA REQUIRED)
+  include_directories(${CUDA_INCLUDE_DIRS})
+  link_directories(${CUDA_LIBRARIES})
   message(STATUS "Using CUDA ${CUDA_VERSION}")
-  set(CUDA_NVCC_FLAGS -std=c++11)
+  set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -std=c++11")
 
   if(CUDA_VERSION VERSION_GREATER "10.1" OR CUDA_VERSION VERSION_EQUAL "10.1")
     add_definitions(-DTRITON_ENABLE_CUDA_GRAPH=1)
@@ -110,8 +97,6 @@ if(${TRITON_ENABLE_GPU})
   cuda_add_library(
     kernel-library-new
     src/kernel.cu src/kernel.h
-    OPTIONS -arch compute_53
-    OPTIONS -code compute_53,sm_53,sm_60,sm_61,sm_62,sm_70,sm_72,sm_75
     OPTIONS -Xcompiler ${HOST_COMPILER_FLAGS}
   )
 endif() # TRITON_ENABLE_GPU
@@ -171,7 +156,7 @@ set_target_properties(
   triton-backend-utils PROPERTIES
   WINDOWS_EXPORT_ALL_SYMBOLS TRUE
   POSITION_INDEPENDENT_CODE ON
-  OUTPUT_NAME tritonbackendutils
+  OUTPUT_NAME triton-backend-utils
 )
 
 target_link_libraries(
@@ -187,7 +172,7 @@ if(${TRITON_ENABLE_GPU})
   target_link_libraries(
     triton-backend-utils
     PUBLIC
-      CUDA::cudart
+      cudart
     PRIVATE
     kernel-library-new
   )
