From 7a9143270124321baa5d3579d5996e2b3789bed9 Mon Sep 17 00:00:00 2001
From: Roger Knecht <roger@norberthealth.com>
Date: Fri, 24 Jun 2022 19:13:11 +0200
Subject: [PATCH] fix cmake build

---
 CMakeLists.txt | 25 ++++++-------------------
 1 file changed, 6 insertions(+), 19 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 622f69a..1fa4c8f 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -24,7 +24,7 @@
 # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 
-cmake_minimum_required(VERSION 3.17)
+cmake_minimum_required(VERSION 3.16)
 
 project(tritonbackend LANGUAGES C CXX)
 
@@ -45,28 +45,15 @@ endif()
 #
 # Dependencies
 #
-include(FetchContent)
-
-FetchContent_Declare(
-  repo-common
-  GIT_REPOSITORY https://github.com/triton-inference-server/common.git
-  GIT_TAG ${TRITON_COMMON_REPO_TAG}
-  GIT_SHALLOW ON
-)
-FetchContent_Declare(
-  repo-core
-  GIT_REPOSITORY https://github.com/triton-inference-server/core.git
-  GIT_TAG ${TRITON_CORE_REPO_TAG}
-  GIT_SHALLOW ON
-)
-FetchContent_MakeAvailable(repo-common repo-core)
 
 #
 # CUDA
 #
 if(${TRITON_ENABLE_GPU})
-  find_package(CUDAToolkit REQUIRED)
+  enable_language(CUDA)
   find_package(CUDA REQUIRED)
+  include_directories(${CUDA_INCLUDE_DIRS})
+  link_directories(${CUDA_LIBRARIES})
   message(STATUS "Using CUDA ${CUDA_VERSION}")
   set(CUDA_NVCC_FLAGS -std=c++11)
 
@@ -171,7 +158,7 @@ set_target_properties(
   triton-backend-utils PROPERTIES
   WINDOWS_EXPORT_ALL_SYMBOLS TRUE
   POSITION_INDEPENDENT_CODE ON
-  OUTPUT_NAME tritonbackendutils
+  OUTPUT_NAME triton-backend-utils
 )
 
 target_link_libraries(
@@ -187,7 +174,7 @@ if(${TRITON_ENABLE_GPU})
   target_link_libraries(
     triton-backend-utils
     PUBLIC
-      CUDA::cudart
+      cudart
     PRIVATE
     kernel-library-new
   )
-- 
2.17.1

