From a9f592e26badc49a39fb6a032d156ececa46d2ef Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Thu, 20 Nov 2025 20:08:10 +0000
Subject: [PATCH] Updates for OE cross builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 CMakeLists.txt      | 173 ++++----------------------------------------
 rapids_config.cmake |  14 +++-
 2 files changed, 26 insertions(+), 161 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index f5a878e..c842290 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,140 +12,7 @@
 # the License.
 # =============================================================================
 
-cmake_minimum_required(VERSION 3.30.4 FATAL_ERROR)
-
-include(rapids_config.cmake)
-include(rapids-cmake)
-include(rapids-cpm)
-include(rapids-export)
-
-# Fetch fmt
-function(get_fmt)
-  set(to_install OFF)
-  if(INSTALL_EXPORT_SET IN_LIST ARGN)
-    set(to_install ON)
-  endif()
-
-  include("${rapids-cmake-dir}/cpm/find.cmake")
-  set(version 11.0.2)
-  rapids_cpm_find(
-    fmt "${version}" ${ARGN}
-    GLOBAL_TARGETS fmt::fmt fmt::fmt-header-only
-    CPM_ARGS
-    GIT_REPOSITORY https://github.com/fmtlib/fmt.git
-    GIT_TAG "v${version}"
-    GIT_SHALLOW ON
-    EXCLUDE_FROM_ALL ON
-    OPTIONS "FMT_INSTALL ${to_install}" "CMAKE_POSITION_INDEPENDENT_CODE ON"
-  )
-
-  # Propagate up variables that CPMFindPackage provide
-  set(fmt_SOURCE_DIR
-      "${fmt_SOURCE_DIR}"
-      PARENT_SCOPE
-  )
-  set(fmt_BINARY_DIR
-      "${fmt_BINARY_DIR}"
-      PARENT_SCOPE
-  )
-  set(fmt_ADDED
-      "${fmt_ADDED}"
-      PARENT_SCOPE
-  )
-  set(fmt_VERSION
-      ${version}
-      PARENT_SCOPE
-  )
-endfunction()
-
-# Fetch spdlog
-function(get_spdlog)
-  set(options)
-  set(one_value FMT_OPTION BUILD_EXPORT_SET INSTALL_EXPORT_SET)
-  set(multi_value)
-  cmake_parse_arguments(_RAPIDS "${options}" "${one_value}" "${multi_value}" ${ARGN})
-
-  # Fix up _RAPIDS_UNPARSED_ARGUMENTS to have EXPORT_SETS as this is need for rapids_cpm_find. Also
-  # propagate the user provided build and install export sets.
-  if(_RAPIDS_INSTALL_EXPORT_SET)
-    list(APPEND _RAPIDS_UNPARSED_ARGUMENTS INSTALL_EXPORT_SET ${_RAPIDS_INSTALL_EXPORT_SET})
-  endif()
-  if(_RAPIDS_BUILD_EXPORT_SET)
-    list(APPEND _RAPIDS_UNPARSED_ARGUMENTS BUILD_EXPORT_SET ${_RAPIDS_BUILD_EXPORT_SET})
-  endif()
-
-  set(to_install OFF)
-  if(_RAPIDS_INSTALL_EXPORT_SET)
-    set(to_install ON)
-  endif()
-
-  # If the option wasn't passed to the command, default to header only fmt
-  if(NOT _RAPIDS_FMT_OPTION)
-    set(_RAPIDS_FMT_OPTION "EXTERNAL_FMT_HO")
-  endif()
-
-  if(_RAPIDS_FMT_OPTION STREQUAL "BUNDLED")
-    set(spdlog_fmt_option "")
-  elseif(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT")
-    set(spdlog_fmt_option "SPDLOG_FMT_EXTERNAL ON")
-    set(spdlog_fmt_target fmt::fmt)
-  elseif(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT_HO")
-    set(spdlog_fmt_option "SPDLOG_FMT_EXTERNAL_HO ON")
-    set(spdlog_fmt_target fmt::fmt-header-only)
-  elseif(_RAPIDS_FMT_OPTION STREQUAL "STD_FORMAT")
-    set(spdlog_fmt_option "SPDLOG_USE_STD_FORMAT ON")
-  else()
-    message(
-      FATAL_ERROR
-        "Invalid option used for FMT_OPTION, got: ${_RAPIDS_FMT_OPTION}, expected one of: 'BUNDLED', 'EXTERNAL_FMT', 'EXTERNAL_FMT_HO', 'STD_FORMAT'"
-    )
-  endif()
-
-  if(_RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT" OR _RAPIDS_FMT_OPTION STREQUAL "EXTERNAL_FMT_HO")
-    include("${rapids-cmake-dir}/cpm/fmt.cmake")
-
-    # Using `spdlog_ROOT` needs to cause any internal find calls in `spdlog-config.cmake` to first
-    # search beside it before looking globally.
-    list(APPEND fmt_ROOT ${spdlog_ROOT})
-
-    get_fmt(${_RAPIDS_UNPARSED_ARGUMENTS})
-  endif()
-
-  include("${rapids-cmake-dir}/cpm/find.cmake")
-  set(version 1.14.1)
-  rapids_cpm_find(
-    spdlog "${version}" ${_RAPIDS_UNPARSED_ARGUMENTS}
-    GLOBAL_TARGETS spdlog::spdlog spdlog::spdlog_header_only
-    CPM_ARGS
-    GIT_REPOSITORY https://github.com/gabime/spdlog.git
-    GIT_TAG "v${version}"
-    GIT_SHALLOW ON
-    EXCLUDE_FROM_ALL ON
-    OPTIONS "SPDLOG_INSTALL ${to_install}" "${spdlog_fmt_option}"
-  )
-
-  # Propagate up variables that CPMFindPackage provide
-  set(spdlog_SOURCE_DIR
-      "${spdlog_SOURCE_DIR}"
-      PARENT_SCOPE
-  )
-  set(spdlog_BINARY_DIR
-      "${spdlog_BINARY_DIR}"
-      PARENT_SCOPE
-  )
-  set(spdlog_ADDED
-      "${spdlog_ADDED}"
-      PARENT_SCOPE
-  )
-  set(spdlog_VERSION
-      ${version}
-      PARENT_SCOPE
-  )
-  set(spdlog_fmt_target
-      ${spdlog_fmt_target}
-      PARENT_SCOPE
-  )
-endfunction()
+cmake_minimum_required(VERSION 3.28.3 FATAL_ERROR)
 
 file(READ "${CMAKE_CURRENT_LIST_DIR}/VERSION" _version)
 string(STRIP "${_version}" _version)
@@ -157,6 +24,18 @@ project(
   LANGUAGES CXX
 )
 
+include(rapids_config.cmake)
+include(rapids-cmake)
+include(rapids-cpm)
+include(rapids-export)
+
+# Fetch fmt
+find_package(PkgConfig)
+pkg_check_modules(FMT REQUIRED IMPORTED_TARGET GLOBAL fmt>=11.0.2)
+
+# Fetch spdlog
+find_package(spdlog 1.16.0 REQUIRED)
+
 rapids_cmake_build_type(Release)
 
 rapids_cpm_init()
@@ -197,32 +76,6 @@ set_target_properties(
              POSITION_INDEPENDENT_CODE ON
 )
 
-# Get spdlog
-if(RAPIDS_LOGGER_HIDE_ALL_SPDLOG_SYMBOLS)
-  set(CPM_DOWNLOAD_spdlog ON)
-  get_spdlog(
-    FMT_OPTION ${RAPIDS_LOGGER_FMT_OPTION} CPM_ARGS OPTIONS "BUILD_SHARED_LIBS OFF"
-    "SPDLOG_BUILD_SHARED OFF"
-  )
-  set_target_properties(spdlog PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  target_link_options(rapids_logger PRIVATE "LINKER:--exclude-libs,libspdlog")
-else()
-  get_spdlog(
-    FMT_OPTION
-    ${RAPIDS_LOGGER_FMT_OPTION}
-    BUILD_EXPORT_SET
-    rapids-logger-exports
-    INSTALL_EXPORT_SET
-    rapids-logger-exports
-    CPM_ARGS
-    OPTIONS
-    "BUILD_SHARED_LIBS ${BUILD_SHARED_LIBS}"
-    "SPDLOG_BUILD_SHARED ${BUILD_SHARED_LIBS}"
-  )
-  if(spdlog_ADDED)
-    set_target_properties(spdlog PROPERTIES POSITION_INDEPENDENT_CODE ON)
-  endif()
-endif()
 target_link_libraries(rapids_logger PRIVATE spdlog::spdlog)
 
 if(BUILD_TESTS)
diff --git a/rapids_config.cmake b/rapids_config.cmake
index 75f5624..535ab5f 100644
--- a/rapids_config.cmake
+++ b/rapids_config.cmake
@@ -26,10 +26,22 @@ else()
   )
 endif()
 
+set(CPM_DOWNLOAD_VERSION 0.40.2)
+set(CPM_DOWNLOAD_LOCATION "${CPM_SOURCE_CACHE}/cpm/CPM_${CPM_DOWNLOAD_VERSION}.cmake")
+message(STATUS "CPM download location: ${CPM_DOWNLOAD_LOCATION}")
+
+if(NOT EXISTS "${RAPIDS_CMAKE_DIR}/RAPIDS.cmake")
+  message(FATAL_ERROR "Required RAPIDS.cmake does not exist!")
+endif()
+
+set(rapids-cmake_POPULATED "TRUE")
 if(NOT rapids-cmake-version)
   set(rapids-cmake-version "${RAPIDS_VERSION_MAJOR_MINOR}")
 endif()
 if(NOT rapids-cmake-branch)
   set(rapids-cmake-branch "branch-25.10")
 endif()
-include("${CMAKE_CURRENT_LIST_DIR}/cmake/RAPIDS.cmake")
+set(rapids-cmake-dir "${RAPIDS_CMAKE_DIR}/rapids-cmake")
+message(STATUS "Rapids cmake DIR: ${rapids-cmake-dir}")
+list(APPEND CMAKE_MODULE_PATH "${rapids-cmake-dir}")
+include("${RAPIDS_CMAKE_DIR}/RAPIDS.cmake")
-- 
2.34.1

