From 1948d905d21b1608530047d0445d844ccb9f31fb Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Thu, 11 Sep 2025 18:38:18 +0100
Subject: [PATCH 3/7] Build python libs with install RPATH

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ian Stewart <istewart@nvidia.com>
---
 python/holoscan/CMakeLists.txt | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/python/holoscan/CMakeLists.txt b/python/holoscan/CMakeLists.txt
index 635b3e9..6c1c75d 100644
--- a/python/holoscan/CMakeLists.txt
+++ b/python/holoscan/CMakeLists.txt
@@ -73,9 +73,8 @@ function(holoscan_pybind11_module pybind11_module_name)
         "\$ORIGIN/${wheel_lib_relative_path}"   # in our python wheel
     )
     list(JOIN _rpath ":" _rpath)
-    set_property(TARGET ${module_target}
-        APPEND PROPERTY BUILD_RPATH ${_rpath}
-    )
+    set_target_properties(${module_target} PROPERTIES INSTALL_RPATH ${_rpath})
+    set_target_properties(${module_target} PROPERTIES BUILD_WITH_INSTALL_RPATH ON)
     unset(_rpath)
 
     # make submodule folder
-- 
2.34.1

