From bd0a5e1da718cce4c37b1fbaaaecc07bcfa71f9e Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Tue, 24 Jun 2025 11:02:49 +0100
Subject: [PATCH 5/7] Disable various warnings as errors

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ian Stewart <istewart@nvidia.com>
---
 CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index d0f3c65..1cee9b1 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -106,6 +106,18 @@ set(CMAKE_CXX_EXTENSIONS NO)
 # Don't append suffix to shared libraries for debug build
 set(CMAKE_DEBUG_POSTFIX "")
 
+# Disable various warnings as errors
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=dangling-pointer>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=dangling-reference>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=deprecated-declarations>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=ignored-attributes>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=ignored-qualifiers>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=missing-field-initializers>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=narrowing>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=overloaded-virtual>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=pessimizing-move>)
+add_compile_options($<$<COMPILE_LANGUAGE:CXX>:-Wno-error=unused-parameter>)
+
 # ##############################################################################
 # # RAPIDS CMAKE configurations
 # # (https://docs.rapids.ai/api/rapids-cmake/stable/api.html)
-- 
2.34.1

