From 3a337764c70a0f55f5b47233e13e52f041c30c6e Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Mon, 28 Apr 2025 16:15:58 +0100
Subject: [PATCH] Updates for OE cross builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 CMakeLists.txt | 50 +++++++++++++++++++++++++++-----------------------
 1 file changed, 27 insertions(+), 23 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 369944c..68cacaf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,29 +1,33 @@
-cmake_minimum_required(VERSION 3.10)
-project(SafeInt VERSION 3.0.26)
+cmake_minimum_required(VERSION 3.0)
 
-set(CMAKE_CXX_STANDARD 11)
-set(CMAKE_CXX_STANDARD_REQUIRED True)
+project(SafeInt CXX)
 
-# Runtime tests are:
-# - default
-# - without built-in 128-bit support
-# - without intrinsics
+if(POLICY CMP0054)
+    cmake_policy(SET CMP0054 NEW)
+endif()
 
-# Compile time tests are:
-# - default C++11
-# - C++14
-# - TODO - consider adding in 17, 20 just to see if anything breaks
-# - compile without exceptions
+option(SAFEINT_TESTS "compile the tests" OFF)
+option(SAFEINT_NO_EXCEPTIONS "disable exceptions" OFF)
+option(SAFEINT_SANITIZE "sanitize undefined behaviour" OFF)
 
-# Supported compilers:
-# - Microsoft
-# - clang
-# - gcc
-# - Intel (not regularly tested)
-# other compilers on a best effort
+add_library(SafeInt INTERFACE)
+target_include_directories(SafeInt INTERFACE .)
+install(FILES SafeInt.hpp DESTINATION include)
+install(FILES LICENSE DESTINATION share)
 
-add_subdirectory(Test/ClangTest)
-add_subdirectory(Test/GccTest)
-add_subdirectory(Test/MsvcTest)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+    if(MSVC_VERSION LESS 1900)
+        # Visual Studio compilers prior to 2015 don't support noexcept:
+        target_compile_definitions(SafeInt INTERFACE SAFEINT_REMOVE_NOTHROW)
+    endif()
+endif()
 
-enable_testing()
\ No newline at end of file
+if(SAFEINT_NO_EXCEPTIONS)
+    if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        target_compile_options(SafeInt INTERFACE -fno-exceptions)
+    endif()
+endif()
+
+if(SAFEINT_TESTS)
+    add_subdirectory(Test)
+endif()
-- 
2.34.1

