From 2a53b45e0af2ff3d179e543c8bec07d80642e7d5 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Tue, 9 Dec 2025 20:20:24 +0000
Subject: [PATCH 8/9] Remove native CUDA_ARCHITECTURE

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ian Stewart <istewart@nvidia.com>
---
 operators/tool_tracking_postprocessor/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/operators/tool_tracking_postprocessor/CMakeLists.txt b/operators/tool_tracking_postprocessor/CMakeLists.txt
index a288d518..4ecda911 100644
--- a/operators/tool_tracking_postprocessor/CMakeLists.txt
+++ b/operators/tool_tracking_postprocessor/CMakeLists.txt
@@ -30,8 +30,6 @@ set_target_properties(tool_tracking_postprocessor
   PROPERTIES
     # separable compilation is required since we launch kernels from within kernels
     CUDA_SEPARABLE_COMPILATION ON
-    # compile for the architecture of the current GPU
-    CUDA_ARCHITECTURES "native"
   )
 
 target_link_libraries(tool_tracking_postprocessor
-- 
2.34.1

