From 736b3e5f0fdb8b7d8218f096d80a9e5935185dd9 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Thu, 22 Jan 2026 12:43:01 -0800
Subject: [PATCH 8/9] Remove native CUDA_ARCHITECTURE

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ian Stewart <istewart@nvidia.com>
---
 operators/tool_tracking_postprocessor/CMakeLists.txt | 2 --
 1 file changed, 2 deletions(-)

diff --git a/operators/tool_tracking_postprocessor/CMakeLists.txt b/operators/tool_tracking_postprocessor/CMakeLists.txt
index a288d518..4ecda911 100644
--- a/operators/tool_tracking_postprocessor/CMakeLists.txt
+++ b/operators/tool_tracking_postprocessor/CMakeLists.txt
@@ -30,8 +30,6 @@ set_target_properties(tool_tracking_postprocessor
   PROPERTIES
     # separable compilation is required since we launch kernels from within kernels
     CUDA_SEPARABLE_COMPILATION ON
-    # compile for the architecture of the current GPU
-    CUDA_ARCHITECTURES "native"
   )
 
 target_link_libraries(tool_tracking_postprocessor
-- 
2.43.0

