From 1463bd9e2c9e1eed3c45fb59cc44579d045faf77 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Mon, 28 Apr 2025 12:29:34 +0100
Subject: [PATCH] Export cmake config file

Upstream-Status: Inappropriate [OE-specific]

Signed-off-by: Kurt Kiefer <kekiefer@gmail.com>
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b1f1dc..e6a7576 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -403,6 +403,7 @@ install (TARGETS nsync
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Development)
 
 install (TARGETS nsync_cpp OPTIONAL
+	EXPORT nsyncConfig
 	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT RuntimeLibraries
 	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT Development)
 
@@ -427,3 +428,5 @@ foreach (NSYNC_INCLUDE ${NSYNC_INCLUDES})
 		DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 		COMPONENT Development)
 endforeach ()
+
+install(EXPORT nsyncConfig NAMESPACE nsync:: DESTINATION ${CMAKE_INSTALL_LIBDIR}/cmake/nsync)
-- 
2.34.1

