From e70e3b340a0e0a0b35ab0a66a6629bc571931fb7 Mon Sep 17 00:00:00 2001
From: Ilies CHERGUI <ichergui@nvidia.com>
Date: Wed, 30 Apr 2025 14:28:25 +0100
Subject: [PATCH] Updates for OE cross builds

Upstream-Status: Inappropriate [OE-specific]
Signed-off-by: Ilies CHERGUI <ichergui@nvidia.com>
---
 cub/cmake/CubBuildTargetList.cmake | 1 -
 1 file changed, 1 deletion(-)

diff --git a/cub/cmake/CubBuildTargetList.cmake b/cub/cmake/CubBuildTargetList.cmake
index 426eee5e2..5c6e7f6ac 100644
--- a/cub/cmake/CubBuildTargetList.cmake
+++ b/cub/cmake/CubBuildTargetList.cmake
@@ -207,7 +207,6 @@ function(cub_build_target_list)
 
   # Set up the CUB target while testing out our find_package scripts.
   find_package(CUB REQUIRED CONFIG
-    NO_DEFAULT_PATH # Only check the explicit path in HINTS:
     HINTS "${CUB_SOURCE_DIR}"
   )
 
-- 
2.34.1

